### DISCUSSION TITLE
C# : Add events for node signals

### DISCUSSION THREAD

**Discussion by n-tiz** [godotengine/godot-proposals#4363] [Scripting]
I don't know if it's planned or not, but i think it could be very interesting to define godot signal as C# event, for multiple reason:

- Using the connect method can be very frustrating without auto-completion to find the signal name.
- Writing the correct method also for the same reason.
- The C# already have a nice built-in way to handle events.
- The name of signals and method could not be wrong (typo or code change).

Here is an exemple:
`Button.Connect("pressed", this, "OnPressed"); `
could be write as
`Button.Pressed += OnPressed;`

**Comment by Calinou** [comment]
> - Using the connect method can be very frustrating without auto-completion to find the signal name.
> - Writing the correct method also for the same reason.

Signals and Callables are first-class types in Godot 4.0 (including in GDScript), so this should no longer be an issue.

I also remember C# events already being implemented in 4.0, but my memory may be hazy.

**Reply by markdibarry** [reply]
For reference: They were partly implemented [2 years ago,](https://godotengine.org/article/csharp-ios-signals-events) but no work has been done on them since.
